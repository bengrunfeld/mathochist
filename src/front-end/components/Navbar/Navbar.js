import React from 'react'

const Navbar = () => (
  <div>
    <p>Navbar</p>
  </div>
)

export default Navbar