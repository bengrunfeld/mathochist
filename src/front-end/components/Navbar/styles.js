import styled from 'styled-components'
import { Link } from 'react-router-dom'

const Logo = styled(Link)`
  font-family: comfortaa, helvetica;
  font-size: 20px !important;
  color: white;
`

const NavbarContainer = styled.nav`
  background: #0b5394 !important;
  height: 45px;
`

export { Logo, NavbarContainer }
