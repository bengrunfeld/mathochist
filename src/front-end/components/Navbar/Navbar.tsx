import React from 'react'

import { NavbarContainer, Logo } from './styles'

const Navbar = () => (
  <NavbarContainer className='navbar navbar-expand-lg'>
    <Logo className='navbar-brand ml-2' to='/'>
      Mathochist
    </Logo>
  </NavbarContainer>
)

export default Navbar
