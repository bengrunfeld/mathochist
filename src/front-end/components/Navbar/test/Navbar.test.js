import React from 'react'
import Enzyme, { shallow, mount, render } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'
import Navbar from '../Navbar'
import setUpTree from '../../../../test/setUpTree'


describe('<Navbar />', () => {
  test('matches the snapshot', () => {
    const tree = setUpTree(Navbar, MemoryRouter, mount, '/')
    expect(tree).toMatchSnapshot()
  })
})
