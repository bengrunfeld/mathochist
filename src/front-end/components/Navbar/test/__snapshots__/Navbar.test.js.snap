// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> matches the snapshot 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
        "pathname": "/",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Navbar>
      <nav
        className="navbar navbar-expand-lg navbar-dark bg-dark"
      >
        <Styled(Link)
          className="navbar-brand ml-2"
          to="/"
        >
          <StyledComponent
            className="navbar-brand ml-2"
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": true,
                  "lastClassName": "buxIHh",
                  "rules": Array [
                    "
  font-family: comfortaa, helvetica;
  font-size: 26px !important;
",
                  ],
                },
                "displayName": "Styled(Link)",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": [Function],
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            to="/"
          >
            <Link
              className="navbar-brand ml-2 sc-bdVaJa buxIHh"
              to="/"
            >
              <a
                className="navbar-brand ml-2 sc-bdVaJa buxIHh"
                href="/"
                onClick={[Function]}
              >
                Ben Grunfeld
              </a>
            </Link>
          </StyledComponent>
        </Styled(Link)>
        <button
          aria-controls="navbarSupportedContent"
          aria-expanded="false"
          aria-label="Toggle navigation"
          className="navbar-toggler"
          data-target="#navbarSupportedContent"
          data-toggle="collapse"
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          id="navbarSupportedContent"
        >
          <ul
            className="navbar-nav mr-auto d-lg-none"
          >
            <li
              className="nav-item active"
            >
              <a
                className="nav-link"
                href="#"
              >
                Home 
                <span
                  className="sr-only"
                >
                  (current)
                </span>
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href="#"
              >
                Link
              </a>
            </li>
            <li
              className="nav-item dropdown"
            >
              <a
                aria-expanded="false"
                aria-haspopup="true"
                className="nav-link dropdown-toggle"
                data-toggle="dropdown"
                href="#"
                id="navbarDropdown"
                role="button"
              >
                Dropdown
              </a>
              <div
                aria-labelledby="navbarDropdown"
                className="dropdown-menu"
              >
                <a
                  className="dropdown-item"
                  href="#"
                >
                  Action
                </a>
                <a
                  className="dropdown-item"
                  href="#"
                >
                  Another action
                </a>
                <div
                  className="dropdown-divider"
                />
                <a
                  className="dropdown-item"
                  href="#"
                >
                  Something else here
                </a>
              </div>
            </li>
            <li
              className="nav-item"
            >
              <a
                className="nav-link disabled"
                href="#"
              >
                Disabled
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </Navbar>
  </Router>
</MemoryRouter>
`;
